internal/patroni/api.go

```diff
@@ -39,13 +39,17 @@ var _ API = Executor(nil)
 // waits up to two "loop_wait" or until an error occurs. When Patroni is paused,
 // next cannot be blank. Similar to the "POST /switchover" REST endpoint.
 func (exec Executor) ChangePrimaryAndWait(
-	ctx context.Context, current, next string,
+	ctx context.Context, current, next string, patroniVer4 bool,
 ) (bool, error) {
 	var stdout, stderr bytes.Buffer
 
-	err := exec(ctx, nil, &stdout, &stderr,
-		"patronictl", "switchover", "--scheduled=now", "--force",
-		"--master="+current, "--candidate="+next)
+	// K8SPG-648: patroni v4.0.0 deprecated "master" role.
+	//            We should use "primary" instead
+	cmd := []string{"patronictl", "switchover", "--scheduled=now", "--force", "--primary=" + current, "--candidate=" + next}
+	if !patroniVer4 {
+		cmd = []string{"patronictl", "switchover", "--scheduled=now", "--force", "--master=" + current, "--candidate=" + next}
```

Comment:
rather than duplicating the slice, can we do smth like `cmd[4] = "--master=" + current`?

Reply:
https://github.com/percona/percona-postgresql-operator/pull/921/commits/e070313a3e004aabd4b89cebc5c6c86b4057fcdf